dex/src/main/scala/com/wavesplatform/dex/api/http/routes/MatcherApiRoute.scala

```diff
@@ -1033,6 +1037,55 @@ class MatcherApiRoute(
     complete { Json.toJson(orderDb.transactionsByOrder(orderId)) }
   }
 
+  @Path("/debug/config")
+  @ApiOperation(
+    value = "Returns current matcher's configuration",
+    httpMethod = "GET",
+    authorizations = Array(new Authorization(SwaggerDocService.apiKeyDefinitionName)),
+    tags = Array("debug"),
+    response = classOf[HttpResponse]
+  )
+  def getConfig: Route = (path("config") & get & withAuth)  {
+    complete {
+
+      var outputConfig = config.getObject("waves").toConfig
+
+      def pathShouldNotBeDeleted(path: String): Boolean = ("(user|pass|seed|private)".r findFirstIn path).isEmpty
+
+      def modifyConfig(path: String, config: Config): Unit = {
+        val itr = config.root().entrySet().iterator()
+
+        while(itr.hasNext()){
+
+          val key = itr.next().getKey
+          val absolutePath = if(path.length > 0) s"$path.$key" else key
+
+          config.root().get(key).valueType() match {
+            case ConfigValueType.OBJECT => {
+              if(pathShouldNotBeDeleted(absolutePath)) {
+                if(config.getObject(key).size() == 0) {
+                  outputConfig = outputConfig.withoutPath(absolutePath)
+                } else {
+                  modifyConfig(absolutePath, config.getObject(key).toConfig)
+                }
+              }
+            }
+            case _ => if(!pathShouldNotBeDeleted(absolutePath)) outputConfig = outputConfig.withoutPath(absolutePath)
+          }
+        }
+      }
+
+      modifyConfig("", outputConfig)
+
+      val header = HttpHeader.parse("Content-Type", "application/hocon")  match {
+        case ParsingResult.Ok(header, _) => header
+        case ParsingResult.Error(error)  => throw new Exception(s"Unable to convert to HttpHeader: ${error.summary}")
+      }
+
+      HttpResponse(entity = new ConfigOps(outputConfig).rendered, headers = Seq(header))
```

Comment:
Please refer to https://doc.akka.io/docs/akka-http/current/routing-dsl/directives/respond-with-directives/respondWithHeader.html

Reply:
Also you don't need to create an instance for ConfigOps, just import it and use on `outputConfig`